File: assignments/elixir/allergies/allergies_test.exs
Code.load_file("allergies.exs")
ExUnit.start

defmodule AllergiesTest do
  use ExUnit.Case, async: true
  doctest Allergies

  test "no_allergies_at_all" do
    assert [] == Allergies.list(0)
  end

  test "allergic_to_just_eggs" do
    assert ["eggs"] == Allergies.list(1)
  end

  test "allergic_to_just_peanuts" do
    assert ["peanuts"] == Allergies.list(2)
  end

  test "allergic_to_just_strawberries" do
    assert ["strawberries"] == Allergies.list(8)
  end

  test "allergic_to_eggs_and_peanuts" do
    assert ["eggs", "peanuts"] == Allergies.list(3)
  end

  test "allergic_to_more_than_eggs_but_not_peanuts" do
    assert ["eggs", "shellfish"] == Allergies.list(5)
  end

  test "allergic_to_lots_of_stuff" do
    assert ["strawberries", "tomatoes", "chocolate", "pollen", "cats"] == Allergies.list(248)
  end

  test "allergic_to_everything" do
    assert ["eggs", "peanuts", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"] == Allergies.list(255)
  end

  test "no_allergies_means_not_allergic" do
    assert ! Allergies.allergic_to?(0, "peanuts")
    assert ! Allergies.allergic_to?(0, "cats")
    assert ! Allergies.allergic_to?(0, "strawberries")
  end

  test "is_allergic_to_eggs" do
    assert Allergies.allergic_to?(1, "eggs")
  end

  test "allergic_to_eggs_in_addition_to_other_stuff" do
    assert Allergies.allergic_to?(5, "eggs")
  end

  test "ignore_non_allergen_score_parts" do
    assert ["eggs", "shellfish", "strawberries", "tomatoes", "chocolate", "pollen", "cats"] == Allergies.list(509)
  end
end

File: assignments/ruby/allergies/allergies_test.rb
require 'minitest/autorun'
require_relative 'allergies'

class AllergiesTest < MiniTest::Unit::TestCase

  def test_no_allergies_means_not_allergic
    allergies = Allergies.new(0)
    refute allergies.allergic_to?('peanuts')
    refute allergies.allergic_to?('cats')
    refute allergies.allergic_to?('strawberries')
  end

  def test_is_allergic_to_eggs
    skip
    allergies = Allergies.new(1)
    assert allergies.allergic_to?('eggs')
  end

  def test_allergic_to_eggs_in_addition_to_other_stuff
    skip
    allergies = Allergies.new(5)
    assert allergies.allergic_to?('eggs')
    assert allergies.allergic_to?('shellfish')
    refute allergies.allergic_to?('strawberries')
  end

  def test_no_allergies_at_all
    skip
    allergies = Allergies.new(0)
    assert_equal [], allergies.list
  end

  def test_allergic_to_just_eggs
    skip
    allergies = Allergies.new(1)
    assert_equal ['eggs'], allergies.list
  end

  def test_allergic_to_just_peanuts
    skip
    allergies = Allergies.new(2)
    assert_equal ['peanuts'], allergies.list
  end

  def test_allergic_to_eggs_and_peanuts
    skip
    allergies = Allergies.new(3)
    assert_equal ['eggs', 'peanuts'], allergies.list
  end

  def test_allergic_to_lots_of_stuff
    skip
    allergies = Allergies.new(248)
    assert_equal ['strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats'], allergies.list
  end

  def test_allergic_to_everything
    skip
    allergies = Allergies.new(255)
    assert_equal ['eggs', 'peanuts', 'shellfish', 'strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats'], allergies.list
  end

  def test_ignore_non_allergen_score_parts
    skip
    allergies = Allergies.new(509)
    assert_equal ['eggs', 'shellfish', 'strawberries', 'tomatoes', 'chocolate', 'pollen', 'cats'], allergies.list
  end

end
